src/marqo/core/validation/validation.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+from decimal import Decimal
+from typing import Optional, Tuple
+from marqo.tensor_search.models.index_settings import IndexSettings
+from pydantic import ValidationError
+from fastapi.responses import JSONResponse
+import json
+
+
+def validate_settings_object(index_name, settings_json) -> Tuple[int, Optional[str]]:
+    """
+    Validates index settings.
+
+    Returns:
+        A tuple containing an HTTP status code and optionally a dictionary or an error message. 
+        On success, it returns 200 and a dictionary representing the settings.
+        On failure due to an ValidationError, it returns 400 and an error message.
+        For any other exception, it returns 500 and an error message.
+    """
+    try:
+        settings_object = json.loads(settings_json)
+        index_settings = IndexSettings(**settings_object)
+        marqo_request = index_settings.to_marqo_index_request(index_name)
+        # Convert the successful marqo_request (if needed) to a dictionary representation
+        convert_marqo_request_to_dict(marqo_request)
+        return JSONResponse(
+            content={
+                "validated": True,
+                "index": index_name
+            },
+            status_code=200
+        )
+    except ValidationError as e:
+        return JSONResponse(
+            content={
+                "validated": False,
+                "validation_error": str(e),
+                "index": index_name
+            },
+            status_code=400
+        )
+    except Exception as e:
```

Comment:
When do we expect an Exception? If it's unexpected (i.e. indicates a bug) then let's leave it unhandled. We don't return unhandled exception messages in the API since 2.2

Reply:
If unhandled exception like a bug happens, then should we return 5xx ? If you don't return anything, how should the client know what happened?

Reply:
If you don't handle it, it'll be a 500 automatically. The API takes care of that